test_public_network.py

```diff
@@ -112,25 +115,60 @@ def test_public_network_port_security(approach, two_servers_in_same_subnet):
     attacker.assert_run('sudo apt update --allow-releaseinfo-change')
     attacker.assert_run('sudo apt install -y curl ettercap-text-only tcpdump')
 
-    # The attacker starts poisoning the environment
-    mitm = approach['mitm']
-
-    v4 = victim.ip('public', 4)
-    v6 = victim.ip('public', 6)
+    ipv4 = victim.ip('public', 4)
+    ipv6 = victim.ip('public', 6)
+    # Get first IPv6 link local address
+    ll_addr_v6 = ip_interface(victim.output_of(
+        "ip -6 -brief addr show scope link | awk '{ print $3 }' | head -1"
+    )).ip
+    ipv4_gateway = victim.gateway('public', 4)
+    ipv6_gateway = victim.gateway('public', 6)
+    if ip_version == 4:
```

Comment:
The comments in the test_... methods are verbose and a bit like section headers. I would prefer this be done the same here, so it is in line with the rest of the test suite.
```suggestion

    # Use ARP spoofing for IPv4, NDP spoofing for IPv6
    if ip_version == 4:
```